### PR TITLE
MGMT-12553: fixing some typos, changing the cronjob schedule back to 'weekly', formatting the average job duration to be composed of minutes, seconds

### PR DESCRIPTION
**Pull request by danmanor**
1. fixing typos.
2. changing the `cronjob` schedule back to `@weekly`
3. formatting job average duration to be composed of minutes and seconds

### CODE REVIEW

File: src/jobsautoreport/report.py [hunk 1]
```diff
@@ -24,9 +24,14 @@ def get_number_of_jobs_triggered(
 
     def get_average_duration_of_jobs(
         self, from_date: datetime, to_date: datetime
-    ) -> float:
+    ) -> tuple[int, int]:
```
Comment (adriengentil) [inline src/jobsautoreport/report.py:27]: Return a  `timedelta` instead in will allow you to extract the information you need https://docs.python.org/3/library/datetime.html#datetime.timedelta

File: src/jobsautoreport/main.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def main() -> None:
                 "type": "section",
                 "text": {
                     "type": "mrkdwn",
-                    "text": f"• Jobs success rate: _{success_rate}_\n• Number of jobs triggered: _{number_of_jobs_triggered}_\n• Jobs average duration: _{average_job_duration}_",
+                    "text": f"• Jobs success rate: _{success_rate}_\n• Number of jobs triggered: _{number_of_jobs_triggered}_\n• Jobs average duration: _{average_jobs_duration.seconds // 60}_ minutes, and _{average_jobs_duration.seconds % 60}_ seconds",
```
Comment (adriengentil) [inline src/jobsautoreport/main.py:46]:
```suggestion
                    "text": f"• Jobs success rate: _{success_rate}_\n• Number of jobs triggered: _{number_of_jobs_triggered}_\n• Jobs average duration: _{average_jobs_duration.total_seconds() // 60}_ minutes, and _{average_jobs_duration.total_seconds() % 60}_ seconds",
```
`seconds` counts up to one day https://stackoverflow.com/a/51653034

File: src/jobsautoreport/report.py [hunk 3]
```diff
@@ -24,9 +24,9 @@ def get_number_of_jobs_triggered(
 
     def get_average_duration_of_jobs(
         self, from_date: datetime, to_date: datetime
-    ) -> float:
+    ) -> timedelta:
```
Comment (adriengentil) [inline src/jobsautoreport/report.py:27]:
```suggestion
    ) -> Optional[timedelta]:
```

File: src/jobsautoreport/report.py [hunk 4]
```diff
@@ -24,9 +24,9 @@ def get_number_of_jobs_triggered(
...
         jobs_list = self._querier.query_number_of_jobs_triggered(from_date, to_date)
         if len(jobs_list) == 0:
-            return 0
-        avg = sum([job.duration for job in jobs_list]) / len(jobs_list)
-        return avg
+            return timedelta(seconds=0)
```
[the remark this answers is not in the document]
Comment (adriengentil) [reply src/jobsautoreport/report.py:30]: In order to report the correct information, we need to differentiate a zero from no data.

File: src/jobsautoreport/main.py [hunk 5]
```diff
@@ -43,7 +46,7 @@ def main() -> None:
                 "type": "section",
                 "text": {
                     "type": "mrkdwn",
-                    "text": f"• Jobs success rate: _{success_rate}_\n• Number of jobs triggered: _{number_of_jobs_triggered}_\n• Jobs average duration: _{average_job_duration}_",
+                    "text": f"• Jobs success rate: _{success_rate}_\n• Number of jobs triggered: _{number_of_jobs_triggered}_\n• Jobs average duration: _{int(average_jobs_duration.total_seconds()) // 60}_ minutes, and _{int(average_jobs_duration.total_seconds())  % 60}_ seconds",
```
Comment (osherdp) [inline src/jobsautoreport/main.py:49]: this line a big too long to read
can you separate it to multiple lines? (maybe in a similar way to how it displays)

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/jobsautoreport/main.py [hunk 6]
```diff
@@ -25,11 +25,14 @@ def main() -> None:
...
+    reporter = Reporter(querier)
 
-    success_rate = processor.get_success_rate(a_week_ago, now)
-    number_of_jobs_triggered = processor.get_number_of_jobs_triggered(a_week_ago, now)
-    average_job_duration = processor.get_average_duration_of_jobs(a_week_ago, now)
+    success_rate = reporter.get_success_rate(a_week_ago, now)
+    number_of_jobs_triggered = reporter.get_number_of_jobs_triggered(a_week_ago, now)
+    average_jobs_duration = reporter.get_average_duration_of_jobs(a_week_ago, now)
+
+    if average_jobs_duration is None:
```
Comment (adriengentil) [inline src/jobsautoreport/main.py:34]: it could be better to adapt the message instead and have a log message. Let's fix that later.

**Review by adriengentil — APPROVED** [review]

File: src/jobsautoreport/main.py [hunk 7]
```diff
@@ -43,7 +46,11 @@ def main() -> None:
                 "type": "section",
                 "text": {
                     "type": "mrkdwn",
-                    "text": f"• Jobs success rate: _{success_rate}_\n• Number of jobs triggered: _{number_of_jobs_triggered}_\n• Jobs average duration: _{average_job_duration}_",
+                    "text": f"""
+• Jobs success rate: _{success_rate}_
+• Number of jobs triggered: _{number_of_jobs_triggered}_
+• Jobs average duration: _{int(average_jobs_duration.total_seconds()) // 60}_ minutes, and _{int(average_jobs_duration.total_seconds())  % 60}_ seconds
+""",
```
Comment (adriengentil) [inline src/jobsautoreport/main.py:53]:
```suggestion
                    "text": (f"• Jobs success rate: _{success_rate}_\n"
                                  f"• Number of jobs triggered: _{number_of_jobs_triggered}_\n"
                                  f"• Jobs average duration: _{int(average_jobs_duration.total_seconds()) // 60}_ minutes, and _{int(average_jobs_duration.total_seconds())  % 60}_ seconds")
```
I missed that, I would be cleaner to have multiline in this way as it keeps the indented property,

Comment (adriengentil) [reply src/jobsautoreport/main.py:53]: the suggestion is bad because I cannot indent properly in the comment field :-(

**Pull request merged**